p2p/protocols/protocol.go

```diff
@@ -336,6 +359,19 @@ func (p *Peer) handleIncoming(handle func(ctx context.Context, msg interface{})
 		return errorf(ErrDecode, "<= %v: %v", msg, err)
 	}
 
+	//if the accounting hook is set, call it
+	if p.spec.Hook != nil {
+		if wmsg.Size != uint32(len(wmsg.Payload)) {
```

Comment:
Is this really needed? I am not sure this is a possible case at all.

Reply:
well, the Size field in the wrapped message is totally redundant. that is what we should get rid of